examples/p4est_2d_dgsem/elixir_advection_diffusion_nonperiodic_curved.jl

```diff
@@ -0,0 +1,96 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the linear advection-diffusion equation
+
+diffusivity() = 5.0e-2
+advection_velocity = (1.0, 0.0)
+equations = LinearScalarAdvectionEquation2D(advection_velocity)
+equations_parabolic = LaplaceDiffusion2D(diffusivity(), equations)
+
+# Example setup taken from
+# - Truman Ellis, Jesse Chan, and Leszek Demkowicz (2016).
+#   Robust DPG methods for transient convection-diffusion.
+#   In: Building bridges: connections and challenges in modern approaches
+#   to numerical partial differential equations.
+#   [DOI](https://doi.org/10.1007/978-3-319-41640-3_6).
+function initial_condition_eriksson_johnson(x, t, equations)
+  l = 4
+  epsilon = diffusivity() # TODO: this requires epsilon < .6 due to sqrt
```

Comment:
Should this maybe throw an exception if a user sets the `diffusivity()` too large?

Reply:
If so, we should probably do this for the other elixirs which also use this initial condition. Mind if I move this to an issue?

Reply:
https://github.com/trixi-framework/Trixi.jl/issues/1543

Reply:
I think the `sqrt` below will already throw an error, so I don't think it's too critical, is it?